src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -1235,6 +1235,16 @@ class MacroAssembler: public Assembler {
                        Register z, Register zlen,
                        Register tmp1, Register tmp2, Register tmp3, Register tmp4,
                        Register tmp5, Register tmp6, Register product_hi);
+
+  // CRC32 code for java.util.zip.CRC32::updateBytes() intrinsic.
+  void kernel_crc32(Register crc, Register buf, Register len,
+        Register table0, Register table1, Register table2, Register table3,
+        Register tmp1, Register tmp2, Register tmp3, Register tmp4, Register tmp5);
+  void update_word_crc32(Register crc, Register v, Register tmp1, Register tmp2,
+        Register table0, Register table1, Register table2, Register table3,
+        bool upper);
+  void update_byte_crc32(Register crc, Register val, Register table);
+
```

Comment:
Hi, one thing I'm confused about here, why do these three functions need to be in the `COMPILER2` macro ?

Reply:
I agree to the [OpenJDK Terms of Use](https://openjdk.java.net/legal/tou/) for all comments I make in a project in the [OpenJDK GitHub organization](https://github.com/openjdk).

Reply:
Hello! Thanks for pointing this out, I moved these functions up upwards before `#ifdef COMPILER2`.